type-generator/src/main/java/io/mesosphere/types/dtr/compiler/impl/StringTemplateEngine.java

```diff
@@ -0,0 +1,777 @@
+                        target.put(key, DateTimeFormatter.ISO_LOCAL_DATE_TIME.withZone(this.dateZone));
+                        break;
+                    case "ISO_OFFSET_DATE_TIME":
+                        target.put(key, DateTimeFormatter.ISO_OFFSET_DATE_TIME.withZone(this.dateZone));
+                        break;
+                    case "ISO_ZONED_DATE_TIME":
+                        target.put(key, DateTimeFormatter.ISO_ZONED_DATE_TIME.withZone(this.dateZone));
+                        break;
+                    case "ISO_DATE_TIME":
+                        target.put(key, DateTimeFormatter.ISO_DATE_TIME.withZone(this.dateZone));
+                        break;
+                    case "ISO_ORDINAL_DATE":
+                        target.put(key, DateTimeFormatter.ISO_ORDINAL_DATE.withZone(this.dateZone));
+                        break;
+                    case "ISO_WEEK_DATE":
+                        target.put(key, DateTimeFormatter.ISO_WEEK_DATE.withZone(this.dateZone));
+                        break;
+                    case "ISO_INSTANT":
+                        target.put(key, DateTimeFormatter.ISO_INSTANT.withZone(this.dateZone));
+                        break;
+                    case "RFC_1123_DATE_TIME":
+                        target.put(key, DateTimeFormatter.RFC_1123_DATE_TIME.withZone(this.dateZone));
+                        break;
+                    default:
+                        target.put(key, DateTimeFormatter.ofPattern(value).withZone(this.dateZone));
+                        break;
+                }
+            }
+        });
+    }
+
+    /**
+     * Evaluate the provided template to the given output, using the values for the macros
+     * from the BlockArguments provided
+     *
+     * @param templateName The name of the template to render
+     * @param output       Where to store the generated output
+     * @param arguments    The arguments to the template
+     */
+    private void templateWrite(OutputStream output, String templateName, BlockArguments arguments) throws IOException {
+        // Write template to the output stream
+        byte[] buffer = templateRender(templateName, arguments).getBytes("UTF-8");
+        output.write(buffer);
+    }
+
+    /**
+     * Evaluate the provided template to the given output, using the values for the macros
+     * from the BlockArguments provided
+     *
+     * @param templateName The name of the template to render
+     * @param arguments    The arguments to the template
+     * @return The contents of the template
+     */
+    private String templateRender(String templateName, BlockArguments arguments) throws IOException {
+        ST tpl = group.getInstanceOf(templateName);
+        if (tpl == null) {
+            throw new IOException("Unable to locate template " + templateName);
+        }
+
+        // Import public getter methods as fields from BlockArguments
+        for (Method method : arguments.getClass().getDeclaredMethods()) {
```

Comment:
Isn't StringTemplate doing this by default?

Reply:
Yes, but it does this on the values you define with `.add()`, but it cannot take all the properties you have in "arguments" and expose them as template arguments. 

As a matter of fact I can now see that the `arguments` argument could be simply changed into a `Map<String, BlockArgument>`. (The idea of the `BlockArgument` is to annotate whatever can be passed as an argument to the compiler)